src/main/scala/viper/gobra/ast/internal/theory/TypeHead.scala

```diff
@@ -36,6 +37,7 @@ object TypeHead {
   case object SeqHD extends TypeHead
   case object SetHD extends TypeHead
   case object MSetHD extends TypeHead
+  case object MMapHD extends TypeHead
```

Comment:
I think that `MMapHD` is a bad name. I would use `MathMapHD` or something similar.